### ISSUE TITLE
Edge cases  are inconsistant for rev and fwd fmin and fmax

### ISSUE THREAD

**Issue by t4c1** [stan-dev/math#2060]
## Description
`d fmin(x,y) / d x` is undefined if `x == y`, so fwd returns `NaN`. But in the limits if we are approaching the point `x == y` from either side we get either 0 or 1, one of which is pragmetically returned by rev implemetation. Similar for `fmax`.

I think same handling of edge cases should be used in both implementations. Which one should we use?

## Example
```
var a = 1;
var b = 1;
var c = fmin(a, b);
c.grad(); // a.adj() == 1; b.adj() == 0

fvar d = {1,1};
fvar e = {1,1};
fvar f = fmin(d, e); //f.d() == NaN
```

## Expected Output
Same handling of edge cases for fwd and rev.

#### Current Version:
v3.3.0

**Comment by rok-cesnovar** [comment]
Given this https://www.wolframalpha.com/input/?i=derivative+min%28x%2Cy%29 

![Screenshot from 2020-09-07 09-38-23](https://user-images.githubusercontent.com/28476796/92362118-cddec480-f0ef-11ea-8e6d-9d5f08690b60.png)

it should be how rev does it.

**Comment by andrjohns** [comment]
Closed via #2063
